Sentiment Analysis for Online Reviews

In [6]:
import pandas as pd

a) Downloading, reading and analyzing datasets

In [51]:
# load data in the right format according to readme files
yelp=pd.read_csv("sentiment_labelled_sentences\yelp_labelled.txt",delimiter="\t", names=["Sentence", "Label"])
imdb=pd.read_csv("sentiment_labelled_sentences\imdb_labelled.txt",delimiter="\t", names=["Sentence", "Label"])
amazon=pd.read_csv("sentiment_labelled_sentences\labelled_amazon.txt",delimiter="\t", names=["Sentence", "Label"])

In [52]:
# check if data is balance in all three dataframes

# yelp
onesYelp = len(yelp[yelp['Label'] == 1])
zerosYelp = len(yelp[yelp['Label'] == 0])
print('Number of 1s in Yelp:', onesYelp)
print('Number of 0s in Yelp:', zerosYelp)

#imdb
onesImdb = len(imdb[imdb['Label'] == 1])
zerosImdb = len(imdb[imdb['Label'] == 0])
print('Number of 1s in Imdb:', onesImdb)
print('Number of 0s in Imdb:', zerosImdb)

#amazon
onesAmazon = len(amazon[amazon['Label'] == 1])
zerosAmazon = len(amazon[amazon['Label'] == 0])
print('Number of 1s in Amazon:', onesAmazon)
print('Number of 0s in Amazon:', zerosAmazon)

Number of 1s in Yelp: 500
Number of 0s in Yelp: 500
Number of 1s in Imdb: 386
Number of 0s in Imdb: 362
Number of 1s in Amazon: 500
Number of 0s in Amazon: 500


The data in the Yelp and Amazon files is balanced because there are the same number of 1s and 0s as labels. 
The data in the Imdb file can be considered almost balancen because the number of 1s and 0s is almost the same (386 and 362, respectively). The ratio of 1s to 0s is 386/362 = 1.067.

b) Pre-processing datasets

In [53]:
# convert all letters to lower case
yelp = yelp.applymap(lambda s:s.lower() if type(s) == str else s)
lmdb = lmdb.applymap(lambda s:s.lower() if type(s) == str else s)
amazon = amazon.applymap(lambda s:s.lower() if type(s) == str else s)

For this part, we decided to convert all sentences to lower case, so that the same word with some upper case letters and without them would not be detected as different words since we are using the string type which takes into account there differences. 